atomic_reactor/plugins/pre_check_base_image.py

```diff
@@ -1,79 +1,60 @@
 If this build is an auto-rebuild, use the base image from the image change
 trigger instead of what is in the Dockerfile.
-Tag each image to a unique name (the build name plus a nonce) to be used during
-this build so that it isn't removed by other builds doing clean-up.
 """
 from io import BytesIO
 from typing import Optional, Union
 
-import docker
 import requests
+from osbs.utils import ImageName
+from requests.exceptions import HTTPError, RetryError, Timeout
 
 from atomic_reactor.plugin import PreBuildPlugin
-from atomic_reactor.util import (get_platforms, base_image_is_custom,
+from atomic_reactor.util import (get_platforms, base_image_is_custom, base_image_is_scratch,
                                  get_checksums, get_manifest_media_type,
-                                 RegistrySession, RegistryClient, map_to_user_params)
-from atomic_reactor.utils import imageutil
-from requests.exceptions import HTTPError, RetryError, Timeout
-from osbs.utils import ImageName
+                                 RegistrySession, RegistryClient)
 
 
-class PullBaseImagePlugin(PreBuildPlugin):
-    key = "pull_base_image"
+class CheckBaseImagePlugin(PreBuildPlugin):
+    key = "check_base_image"
     is_allowed_to_fail = False
 
-    args_from_user_params = map_to_user_params("parent_images_digests")
-
-    def __init__(self, workflow, check_platforms=False, inspect_only=False,
-                 parent_images_digests=None):
+    def __init__(self, workflow):
         """
         constructor
 
         :param workflow: DockerBuildWorkflow instance
-        :param check_platforms: validate parent images provide all platforms expected for the build
-        :param inspect_only: bool, if set to True, base images will not be pulled
-        :param parent_images_digests: dict, parent images manifest digests
         """
         # call parent constructor
-        super(PullBaseImagePlugin, self).__init__(workflow)
-
-        self.check_platforms = check_platforms
-        self.inspect_only = inspect_only
-        pull_registries = workflow.conf.pull_registries
+        super(CheckBaseImagePlugin, self).__init__(workflow)
 
         self.source_registry_docker_uri = self.workflow.conf.source_registry['uri'].docker_uri
 
-        self.allowed_registries = [reg['uri'].docker_uri for reg in pull_registries]
+        self.allowed_registries = [reg['uri'].docker_uri for reg in workflow.conf.pull_registries]
         self.allowed_registries.append(self.source_registry_docker_uri)
 
-        if parent_images_digests:
-            # OSBS2 TBD
-            metadata = self.workflow.parent_images_digests
-            metadata.update(parent_images_digests)
-
         self.manifest_list_cache = {}
         # RegistryClient instances cached by registry name
         self.registry_clients = {}
 
     def run(self):
         """
-        Pull parent images and retag them uniquely for this build.
+        Check parent images to ensure they only come from allowed registries.
```

Comment:
This plugin also:

- `_validate_platforms_in_image`
- `_store_manifest_digest`

If you like, I think you can fix the plugin description specifically.

Reply:
updated.